consensus/pow/src/matrix.rs

```diff
@@ -51,16 +51,15 @@ impl Matrix {
 
     #[inline(always)]
     fn convert_to_float(&self) -> [[f64; 64]; 64] {
-        // SAFETY: An uninitialized MaybeUninit is always safe.
-        let mut out: [[MaybeUninit<f64>; 64]; 64] = unsafe { MaybeUninit::uninit().assume_init() };
+        let mut out: [[f64; 64]; 64] = [[Default::default(); 64]; 64];
 
         out.iter_mut().zip(self.0.iter()).for_each(|(out_row, mat_row)| {
             out_row.iter_mut().zip(mat_row).for_each(|(out_element, &element)| {
-                out_element.write(f64::from(element));
+                *out_element = f64::from(element);
             })
         });
-        // SAFETY: The loop above wrote into all indexes.
-        unsafe { std::mem::transmute(out) }
+
+        out
```

Comment:
This will be much slower, this one seems to optimize as well as the original:
`std::array::from_fn(|i| std::array::from_fn(|j| f64::from(self.0[i][j])))`

See profiling(llvm-mca) and assembly here: https://godbolt.org/z/q5aj7chn6